pkg/streamingpromql/types/operator.go

```diff
@@ -10,6 +10,14 @@ import (
 	"github.com/prometheus/prometheus/promql/parser/posrange"
 )
 
+type PrepareParams struct {
+	QueryStats *QueryStats
+}
+
+func (p *PrepareParams) Clone() PrepareParams {
```

Comment:
Given we always work with a pointer to `PrepareParams`, it might be nicer to return a `*PrepareParams` here.

It'd be good to then introduce a test that `Clone` returns an equivalent value (ie. all fields have the same values), but is a different instance of `PrepareParams`.

Reply:
Same as above